server/models.py

```diff
@@ -630,3 +637,34 @@ class Version(Model):
     name = db.Column(db.String(255), nullable=False, unique=True, index=True)
     current_version = db.Column(db.String(255), nullable=False)
     download_link = db.Column(db.Text())
+
+
+class Comment(Model):
+    """ Composition comments. Line is the line # on the Diff.
+    Submission_line is the closest line on the submitted file.
+    """
+    id = db.Column(db.Integer(), primary_key=True)
+    backup_id = db.Column(db.ForeignKey("backup.id"), nullable=False)
+    author_id = db.Column(db.ForeignKey("user.id"), nullable=False)
+
+    filename = db.Column(db.String(255), nullable=False)
+    line = db.Column(db.Integer(), nullable=False) # Line of the original file
+
+    message = db.Column(mysql.MEDIUMTEXT)  # Markdown
+
+    @property
+    def formatted(self):
+        return markdown(self.message)
+
+class Score(Model):
+    id = db.Column(db.Integer, primary_key=True)
+    grader_id = db.Column(db.ForeignKey("user.id"), nullable=False)
+    assignment_id = db.Column(db.ForeignKey("assignment.id"), nullable=False)
+    backup_id = db.Column(db.ForeignKey("backup.id"), nullable=False, index=True)
+
+    kind = db.Column(db.String(255), nullable=False)
+    score = db.Column(db.Float, nullable=False)
+    message = db.Column(mysql.MEDIUMTEXT)
+    public = db.Column(db.Boolean, default=True)
```

Comment:
Does this mean we'll need a giant UPDATE over every score in order to release scores?